File: eio2024-ev/2024-12-08-ev/kook/solution/sol3.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int n;
    cin >> n;

    vector<int> p(n), m(n);
    for (int i = 0; i < n; ++i) {
        cin >> p[i];
    }
    for (int i = 0; i < n; ++i) {
        cin >> m[i];
    }

    vector<int> p_sort = p, m_sort = m;
    sort(p_sort.begin(), p_sort.end());
    sort(m_sort.begin(), m_sort.end());

    bool voimalik = 1;
    for (int i = 1; i < n; ++i) {
        if (m_sort[i]*p_sort[i-1] != m_sort[i-1]*p_sort[i]) {
            voimalik = 0;
            break;
        }
    }

    if (voimalik) {
        cout << "JAH\n";
        vector<int> a(n);

        vector<bool> p_voetud(n, 0), m_voetud(n, 0);

        for (int i = 0; i < n; ++i) {
            int p_vota = -1, m_vota = -1;

            for (int j = 0; j < n; ++j) {
                if (p[j] == p_sort[i]) {
                    p_vota = j;
                    break;
                }
            }

            for (int j = 0; j < n; ++j) {
                if (m[j] == m_sort[i]) {
                    m_vota = j;
                    break;
                }
            }

            a[m_vota] = p_vota+1;

            p[p_vota] = -1;
            m[m_vota] = -1;
        }

        for (int i = 0; i < n; ++i) {
            cout << a[i] << " \n"[i==n-1];
        }
    }
    else {
        cout << "EI\n";
    }

    return 0;
}

File: eio2024-ev/2024-12-08-ev/kook/solution/sol3.py
n = int(input())
p = list(map(int, input().split()))
m = list(map(int, input().split()))

p_sorted = sorted(p)
m_sorted = sorted(m)

voimalik = True

for i in range(1,n):
    if (m_sorted[i] * p_sorted[i-1] != m_sorted[i-1] * p_sorted[i]):
        voimalik = False
        break

if (voimalik):
    print('JAH')

    a = [-1]*n
    p_voetud = [False]*n
    m_voetud = [False]*n

    for i in range(n):
        p_vota = -1
        m_vota = -1

        for j in range(n):
            if (p[j] == p_sorted[i]):
                p_vota = j
                break
        
        for j in range(n):
            if (m[j] == m_sorted[i]):
                m_vota = j
                break
        
        a[m_vota] = p_vota+1

        p[p_vota] = -1
        m[m_vota] = -1
    
    print(*a)
else:
    print('EI')